### DISCUSSION TITLE
How to remove trailing comma?

### DISCUSSION THREAD

**Discussion by michaelxswe** [astral-sh/ruff#9350] [Q&A] [answered]
Before applying the formatter

```
def method(arg1, arg2, arg3, ..., arg10):
```

After applying the formatter

```
def method(
    arg1,
    arg2,
    arg3,
    ...,
    arg10,
)
```

Is there a way to configure the ruff formatter to remove the trailing comma next to arg10?

Maybe by passing an argument to

```
"ruff.format.args": ["remove-trailing-comma"]
```

**Comment by tjkuson** [comment, marked as answer]
The Ruff formatter tries to follow the [Black code style](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html), which adds a trailing comma. A configuration option has not been added to Ruff to change that behaviour (it would have to be a new feature).

**Reply by michaelxswe** [reply]
Thanks for the info!

**Comment by heiner** [comment]
Here's a reason a flag to suppress trailing commas might be useful: It turns `ruff format` into a quite decent json formatting engine. E.g.,

```
cat file.json | ruff format - --isolated --config indent-width=2
```

However, sadly trailing commas aren't legal JSON.

**Reply by mgaitan** [reply]
use jq, dude
